RadixWallet/Features/DerivePublicKeys/DerivePublicKeys.swift

```diff
@@ -63,9 +77,29 @@ public struct DerivePublicKeys: Sendable, FeatureReducer {
 			case device
 			case specific(FactorSource)
 			case specificPrivateHDFactorSource(PrivateHDFactorSource)
+
+			var factorSourceAccessKind: FactorSourceAccess.State.Kind {
+				switch self {
+				case .device:
+					.device
+				case let .specific(source):
+					switch source {
+					case .device:
+						.device
+					case let .ledger(ledger):
+						.ledger(ledger)
+					default:
+						fatalError("Implement")
+					}
+				case .specificPrivateHDFactorSource:
+					// TODO: MATI: Confirm this should use device
+					.device
```

Comment:
yes that is correct, "private HD" means "A have this mnemonic in hand, I'm gonna use that, and then later save it into Keychain and Profile, and since it is then ON MY PHONE, ON MY DEVICE, it is `Device`-FactorSource" :)